test/library_server.cpp

```diff
@@ -711,3 +711,73 @@ TEST_F(LibraryServerTest, catalog_v2_partial_entries)
     "</feed>\n"
   );
 }
+
+#define EXPECT_SEARCH_RESULTS(SEARCH_TERM, RESULT_COUNT, OPDS_ENTRIES)      \
+  {                                                                         \
+    const auto r = zfs1_->GET("/ROOT/catalog/search?q=" SEARCH_TERM);       \
+    EXPECT_EQ(r->status, 200);                                              \
+    EXPECT_EQ(maskVariableOPDSFeedData(r->body),                            \
+      OPDS_FEED_TAG                                                         \
+      "  <id>12345678-90ab-cdef-1234-567890abcdef</id>\n"                   \
+      "  <title>Filtered zims (q=" SEARCH_TERM ")</title>\n"                \
+      "  <updated>YYYY-MM-DDThh:mm:ssZ</updated>\n"                         \
+      "  <totalResults>" #RESULT_COUNT "</totalResults>\n"                  \
+      "  <startIndex>0</startIndex>\n"                                      \
+      "  <itemsPerPage>" #RESULT_COUNT "</itemsPerPage>\n"                  \
+      CATALOG_LINK_TAGS                                                     \
+                                                                            \
+      OPDS_ENTRIES                                                          \
+                                                                            \
+      "</feed>\n"                                                           \
+    );                                                                      \
+  }
+
+TEST_F(LibraryServerTest, catalog_search_includes_public_tags)
+{
+  EXPECT_SEARCH_RESULTS("public_tag_without_a_value",
+                        1,
+                        RAY_CHARLES_CATALOG_ENTRY
+  );
+
+  EXPECT_SEARCH_RESULTS("public_tag_with_a_value",
+                        1,
+                        UNCATEGORIZED_RAY_CHARLES_CATALOG_ENTRY
+  );
+
+  // prefix search works on tag names
+  EXPECT_SEARCH_RESULTS("public_tag",
+                        2,
+                        RAY_CHARLES_CATALOG_ENTRY
+                        UNCATEGORIZED_RAY_CHARLES_CATALOG_ENTRY
+  );
+
+  EXPECT_SEARCH_RESULTS("value_of_a_public_tag",
+                        1,
+                        UNCATEGORIZED_RAY_CHARLES_CATALOG_ENTRY
+  );
+
+  // prefix search works on tag values
+  EXPECT_SEARCH_RESULTS("value_of",
+                        1,
+                        UNCATEGORIZED_RAY_CHARLES_CATALOG_ENTRY
+  );
+}
+
+#define EXPECT_ZERO_RESULTS(SEARCH_TERM) EXPECT_SEARCH_RESULTS(SEARCH_TERM, 0, )
+
+TEST_F(LibraryServerTest, catalog_search_on_tags_is_not_an_any_substring_match)
+{
+  EXPECT_ZERO_RESULTS("tag_with")
+  EXPECT_ZERO_RESULTS("alue_of_a_public_tag")
```

Comment:
It's meant 'value' here right?

Reply:
No. With 'value' the result count shouldn't be zero as checked in another new test added in this PR.

Reply:
ok